Lib/http/server.py

```
@@ -686,12 +687,42 @@ def send_head(self):
         except OSError:
             self.send_error(HTTPStatus.NOT_FOUND, "File not found")
             return None
+
         try:
+            fs = os.fstat(f.fileno())
+            # Use browser cache if possible
+            if ("If-Modified-Since" in self.headers
+                    and "If-None-Match" not in self.headers):
+                # compare If-Modified-Since and time of last file modification
+                try:
+                    ims = email.utils.parsedate_to_datetime(
+                        self.headers["If-Modified-Since"])
+                except:
```

Never use bare `except` unless you reraise the exception. This hides such exceptions as KeyboardInterrupt or MemoryError.

---

I added the `try` block because you said `parsedate_to_datetime()` could raise exceptions ; do you know which ones ? By looking at the code I see that it might raise `TypeError` because `_parsedate_tz` may return `None` and the line
```python
*dtuple, tz = _parsedate_tz(data)
```
would raise a `TypeError` in this case ; but is it the only one ?

Anyway, the code is itself inside a `try...except` block where the `except` block is just `except:`, so this is not making things much worse.

---

That TypeError looks like a bug in the implementation. Other exceptions:

>>> parsedate_to_datetime("Sun, 99 Feb 2017 20:49:31")
ValueError: day is out of range for month
>>> parsedate_to_datetime("Sun, 26 Feb 2017 20:49:31 +12345678901234")
OverflowError: Python int too large to convert to C long
>>> parsedate_to_datetime("Mon, 20 Nov , 19:12:08 GMT")
IndexError: string index out of range

The point is the outer exception handler reraises the exception, but your handler swallows it.

---

Thanks, the exceptions have been specified in commit  1e68be9e1e68e5c483f01043ba0d081a2e538290